src/services/template_converter_service.py

```
@@ -0,0 +1,105 @@
+                capture_output=True,
+                text=True,
+            )
+
+            if temp_pdf_path.exists():
+                pdf_bytes = temp_pdf_path.read_bytes()
+                return pdf_bytes
+            else:
+                raise Exception("PDF generation failed")
+
+        except subprocess.CalledProcessError as e:
+            raise Exception(f"Marp PDF generation failed: {e.stderr}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)